### PR TITLE
Device pagination issue fixed in Python and poweshell scripts.

### PR DESCRIPTION
**Pull request by prasadrao-dell**
1. Device pagination issue fixed in Python and PowerShell scripts.

2. Removed request module dependency from get_device_list.py

### CODE REVIEW

File: Core/Python/get_device_list.py [hunk 1]
```diff
@@ -55,61 +52,76 @@ def __init__(self, session_input, output_details):
...
 
         try:
-            if self.__authenticate_with_ome() is False:
+            AUTH_SUCCESS, HEADERS = self.__authenticate_with_ome()
+            print(HEADERS)
+            if not AUTH_SUCCESS:
+                print("Unable to authenticate with OME .. Check IP/Username/Pwd")
                 return
-        except requests.exceptions.RequestException as auth_ex:
+        except OSError:
```
Comment (RaajeevKalyan) [inline Core/Python/get_device_list.py:62]: Would OSError map to Exceptions from the Requests library? I assumed they were returned on system call failures - wouldn't it be better to map to Exception instead?

File: Core/Python/get_device_list.py [hunk 2]
```diff
@@ -55,61 +52,76 @@ def __init__(self, session_input, output_details):
...
+            if self.__get_device_list(HEADERS) is False:
+                print("Get device list failed.")
                 return
-        except requests.exceptions.RequestException as get_ex:
+        except OSError as get_ex:
             print("Unable to get device list from OME appliance %s" % self.__session_input["ip"])
             print(get_ex)
             return
-
+        print("Get device list passed. ")
```
Comment (RaajeevKalyan) [inline Core/Python/get_device_list.py:75]: Please remove this debug statement

File: Core/Python/get_device_list.py [hunk 3]
```diff
@@ -55,61 +52,76 @@ def __init__(self, session_input, output_details):
...
+            print("Unexpected error:", sys.exc_info())
             return
 
         try:
-            if self.__get_device_list() is False:
+            if self.__get_device_list(HEADERS) is False:
+                print("Get device list failed.")
                 return
-        except requests.exceptions.RequestException as get_ex:
+        except OSError as get_ex:
```
Comment (RaajeevKalyan) [inline Core/Python/get_device_list.py:71]: Duplicate issue -

File: Core/Python/set_power_state.py [hunk 4]
```diff
@@ -186,18 +186,25 @@ def get_execution_detail(job_hist_resp, headers, job_hist_url):
...
+			dev_json_response = device_response.json()
+			if dev_json_response['@odata.count'] <= 0:
+				print("No devices found at ", ip_address)
+				return
+
+			if '@odata.nextLink' in dev_json_response:
+				next_link_url = 'https://%s/' %ip_address + dev_json_response['@odata.nextLink']
+
+			if dev_json_response['@odata.count'] > 0:
+				ome_device_list = [x['Id'] for x in dev_json_response['value']]
```
Comment (RaajeevKalyan) [inline Core/Python/set_power_state.py]: This looks like a bug - shouldn't you be appending to the list? I'd think if this code is executed you'd end up having just the device IDs in the last page of results returned.

File: Core/Python/set_power_state.py [hunk 5]
```diff
@@ -241,12 +248,14 @@ def get_power_states(ip_address, device_id, headers):
 			DEVICE_LIST = get_device_list(IP_ADDRESS, HEADERS)
 			if DEVICE_LIST:
 				if int(DEVICE_ID) in DEVICE_LIST:
+					print("Device found: " + str(DEVICE_ID))
```
Comment (RaajeevKalyan) [inline Core/Python/set_power_state.py:251]: Remove debug statement

File: Core/Python/set_system_configuration.py [hunk 6]
```diff
@@ -51,22 +51,31 @@ def authenticate_with_ome(ip_address, user_name, password):
...
+			dev_json_response = device_response.json()
+			if dev_json_response['@odata.count'] <= 0:
+				print("No devices found at ", ip_address)
+				return
+
+			if '@odata.nextLink' in dev_json_response:
+				next_link_url = 'https://%s/' %ip_address + dev_json_response['@odata.nextLink']
+
+			if dev_json_response['@odata.count'] > 0:
+				ome_device_list = [x['Id'] for x in dev_json_response['value']]
```
Comment (RaajeevKalyan) [inline Core/Python/set_system_configuration.py]: I could be mistaken but this logic seems incorrect - refer prior comments

File: Core/Python/update_firmware_using_dup.py [hunk 7]
```diff
@@ -106,19 +106,27 @@ def get_group_list(ip_address, headers):
...
+			dev_json_response = device_response.json()
+			if dev_json_response['@odata.count'] <= 0:
+				print("No devices found at ", ip_address)
+				return
+
+			if '@odata.nextLink' in dev_json_response:
+				next_link_url = 'https://%s/' %ip_address + dev_json_response['@odata.nextLink']
+
+			if dev_json_response['@odata.count'] > 0:
+				ome_device_list = [x['Id'] for x in dev_json_response['value']]
```
Comment (RaajeevKalyan) [inline Core/Python/update_firmware_using_dup.py]: Check

File: Core/Python/.idea/.gitignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (RaajeevKalyan) [inline Core/Python/.idea/.gitignore]: I don't think any of these files are required / should have made it into the push to the repo

File: Core/Python/get_device_list.py (hunk 1 above)
Comment (prasadrao-dell) [reply Core/Python/get_device_list.py:62]: Done

File: Core/Python/get_device_list.py (hunk 2 above)
Comment (prasadrao-dell) [reply Core/Python/get_device_list.py:75]: Done

File: Core/Python/get_device_list.py (hunk 3 above)
Comment (prasadrao-dell) [reply Core/Python/get_device_list.py:71]: Done

File: Core/Python/set_power_state.py (hunk 4 above)
Comment (prasadrao-dell) [reply Core/Python/set_power_state.py]: This was already fixed in one of the submitted commits

File: Core/Python/set_power_state.py (hunk 5 above)
Comment (prasadrao-dell) [reply Core/Python/set_power_state.py:251]: Done

File: Core/Python/set_system_configuration.py (hunk 6 above)
Comment (prasadrao-dell) [reply Core/Python/set_system_configuration.py]: This was already fixed in one of the submitted commits

File: Core/Python/update_firmware_using_dup.py (hunk 7 above)
Comment (prasadrao-dell) [reply Core/Python/update_firmware_using_dup.py]: This was already fixed in one of the submitted commits

File: Core/Python/.idea/.gitignore (hunk 8 above)
Comment (prasadrao-dell) [reply Core/Python/.idea/.gitignore]: This was already fixed in one of the submitted commits

**Pull request merged**